### PR TITLE
Extract info panel link html logic to portal js, use configuration to de...

### PR DESCRIPTION
**Pull request by anguss00**
...termine list of online resources.

Fixes issue #1613

### CODE REVIEW

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 1]
```diff
@@ -408,44 +408,17 @@ class LayerController {
...
-                    html += "</ul>"
 
-                    responseText = html
+                    responseText = abstractText
                 }
             }
             catch (SAXException e) {
-                log.warn("Error getting formatted metadata, params: ${params}", e)
+                log.warn("Error getting metadata abstract, params: ${params}", e)
                 responseText = "<BR>The metadata record is not available at this time."
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: If we're formatting stuff in the client side, perhaps returning here a HTTP error is adequate and then the client can deal with it?

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 2]
```diff
@@ -408,44 +408,17 @@ class LayerController {
...
             }
             catch (FileNotFoundException e) {
-                log.warn("Error getting formatted metadata, params: ${params}", e)
+                log.warn("Error getting metadata abstract, params: ${params}", e)
                 responseText = "<BR>The metadata record is not available at this time."
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Same here - with returning an error.

File: web-app/js/portal/details/InfoPanel.js [hunk 3]
```diff
@@ -26,20 +26,54 @@ Portal.details.InfoPanel = Ext.extend(Ext.Container, {
...
         Ext.Ajax.request({
             url: metadataUrl,
             scope: this,
             success: function(resp, options) {
-                this.update(resp.responseText, false);
+
+                this._constructInfoTabHtml(resp.responseText, this.layer.parentGeoNetworkRecord.data.onlineResources);
             },
             failure: function(resp) {
                 this.update("<i>" + OpenLayers.i18n('noMetadataMessage') + "</i>", false);
```
Comment (danfruehauf) [inline web-app/js/portal/details/InfoPanel.js]: So this is where you should return the error to the user, in `noMetadataMessage`.

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy (hunk 1 above)
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: In other words - if the client is already handling errors, and this is just a simple proxy request - let the client do it all the way.

Comment (anguss00) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Agreed, should definitely be handled in the client side. Do you suggest adding logging to the failure clause in the Ajax request on the client?

Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Probably yes, wouldn't hurt.

**Comment by anguss00** [comment]
Changes made @danfruehauf

**Pull request merged**